File: C/7_reverse.c
int reverse(int x) {
    long long reverse = 0;
    
    while (x) {
        reverse = reverse * 10 + x % 10;
        x /= 10;
    }
    
    if (reverse > INT_MAX || reverse < INT_MIN)
        return 0;
    else
        return (int)reverse;
}

File: python/7_reverse.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x>=0:
            res = int(str(x)[::-1])
        else:
            res = int('-' + str(x)[:0:-1])
        
        return res if -2147483648 <= res <= 2147483647 else 0
